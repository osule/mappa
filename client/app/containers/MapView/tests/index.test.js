import React from 'react';
import sinon from 'sinon';

import { shallow } from 'enzyme';

import { MapView } from '../index';

describe('<MapView />', () => {
  it('should render a google map', () => {
    const LatLngBounds = () => {};
    LatLngBounds.prototype.getCenter = () => ({ lat: () => 10, lng: () => 20 });
    LatLngBounds.prototype.extend = () => {};
    window.google = {
      maps: {
        LatLngBounds,
        LatLng: sinon.stub(),
        Point: sinon.stub(),
        geometry: {
          spherical: {
            computeHeading: () => 45,
          },
        },
      },
    };
    const markers = {};
    const renderedComponent = shallow(
      <MapView
        initializeMarkers={() => {}}
        markers={markers}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });
});
