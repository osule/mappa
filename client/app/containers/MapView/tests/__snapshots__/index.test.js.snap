// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapView /> should render a google map 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <GoogleMap
        center={
            Object {
                "lat": 10,
                "lng": 20,
              }
        }
        defaultZoom={12}
    >
        
    </GoogleMap>
</div>,
  "nodes": Array [
    <div>
      <GoogleMap
            center={
                  Object {
                        "lat": 10,
                        "lng": 20,
                      }
            }
            defaultZoom={12}
      >
            
      </GoogleMap>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <MapView
        initializeMarkers={[Function]}
        markers={Object {}}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": MapView {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "initializeMarkers": [Function],
          "markers": Object {},
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <GoogleMap
                    center={
                              Object {
                                        "lat": 10,
                                        "lng": 20,
                                      }
                    }
                    defaultZoom={12}
          >
                    
          </GoogleMap>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <GoogleMap
                    center={
                              Object {
                                        "lat": 10,
                                        "lng": 20,
                                      }
                    }
                    defaultZoom={12}
          >
                    
          </GoogleMap>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MapView
    initializeMarkers={[Function]}
    markers={Object {}}
/>,
}
`;
